### PR TITLE
Unit Tests for Event

### PR DESCRIPTION
**Pull request by charansingh83sh**

### CODE REVIEW

File: IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+           {
+               // Logging 
+               services.AddLogging(builder =>
+               {
+                   builder.SetMinimumLevel(Microsoft.Extensions.Logging.LogLevel.Debug);
+                   builder.AddConsole();
+                   builder.AddDebug();
+               });
+               // DBContext
+               services.AddDbContext<EventuateTramDbContext>((provider, o) =>
```
Comment (cer) [inline IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs]: This commit https://github.com/eventuate-examples/eventuate-tram-core-dotnet-examples-basic/commit/9aeea28a26132c5e334d2381c39e4c756b6bd1ec removed `services.AddDbContext<EventuateTramDbContext>`

File: IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using IO.Eventuate.Tram.Local.Kafka.Consumer;
+using IO.Eventuate.Tram.Database;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Hosting;
+
+namespace IO.Eventuate.Tram.Tests.TestHelpers
+{
+    public class TestServicesHost
```
Comment (cer) [inline IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs]: MessageTest and EventTest need different configuration.
Each one should have their own service host.
The common code should be in an abstract super class

File: IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+               {
+                   o.UseSqlServer(TestSettings.EventuateTramDbConnection);
+               });
+               // Kafka Transport
+               services.AddEventuateTramSqlKafkaTransport(TestSettings.EventuateTramDbSchema, TestSettings.KafkaBootstrapServers, EventuateKafkaConsumerConfigurationProperties.Empty(),
+                   (provider, o) =>
+                   {
+                       o.UseSqlServer(TestSettings.EventuateTramDbConnection);
+                   });
+               // Publisher
```
Comment (cer) [inline IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs]: MessageTest doesn't need the events configuration. See above comment about having separate Host classes

**Review by cer — CHANGES_REQUESTED** [review]

File: IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs (hunk 1 above)
Comment (charansingh83sh) [reply IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs]: Removed

File: IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs (hunk 2 above)
Comment (charansingh83sh) [reply IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs]: Added Separate Service host in Unit tests itself

File: IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs (hunk 3 above)
Comment (charansingh83sh) [reply IO.Eventuate.Tram.Tests/TestHelpers/TestServicesHost.cs]: Added Separate Service host in Unit tests itself

**Pull request merged**